src/tokenizer.ts

```diff
@@ -168,182 +177,352 @@ export enum Token {
+  /* 0x00 */ Token.INVALID,
+  /* 0x01 */ Token.INVALID,
+  /* 0x02 */ Token.INVALID,
+  /* 0x03 */ Token.INVALID,
+  /* 0x04 */ Token.INVALID,
+  /* 0x05 */ Token.INVALID,
+  /* 0x06 */ Token.INVALID,
+  /* 0x07 */ Token.INVALID,
+  /* 0x08 */ Token.INVALID,
+  /*   \t */ Token.WHITESPACE,
+  /*   \n */ Token.WHITESPACE,
+  /*   \v */ Token.WHITESPACE,
+  /*   \f */ Token.WHITESPACE,
+  /*   \r */ Token.WHITESPACE,
+  /* 0x0E */ Token.INVALID,
+  /* 0x0F */ Token.INVALID,
+  /* 0x10 */ Token.INVALID,
+  /* 0x11 */ Token.INVALID,
+  /* 0x12 */ Token.INVALID,
+  /* 0x13 */ Token.INVALID,
+  /* 0x14 */ Token.INVALID,
+  /* 0x15 */ Token.INVALID,
+  /* 0x16 */ Token.INVALID,
+  /* 0x17 */ Token.INVALID,
+  /* 0x18 */ Token.INVALID,
+  /* 0x19 */ Token.INVALID,
+  /* 0x1A */ Token.INVALID,
+  /* 0x1B */ Token.INVALID,
+  /* 0x1C */ Token.INVALID,
+  /* 0x1D */ Token.INVALID,
+  /* 0x1E */ Token.INVALID,
+  /* 0x1F */ Token.INVALID,
+  /*  ' ' */ Token.WHITESPACE,
+  /*    ! */ Token.OPERATOR,
+  /*    " */ Token.STRINGLITERAL,
+  /*    # */ Token.INVALID,
+  /*    $ */ Token.IDENTIFIER,
+  /*    % */ Token.OPERATOR,
+  /*    & */ Token.OPERATOR,
+  /*    ' */ Token.STRINGLITERAL,
+  /*    ( */ Token.OPENPAREN,
+  /*    ) */ Token.CLOSEPAREN,
+  /*    * */ Token.OPERATOR,
+  /*    + */ Token.OPERATOR,
+  /*    , */ Token.COMMA,
+  /*    - */ Token.OPERATOR,
+  /*    . */ Token.OPERATOR,
+  /*    / */ Token.COMMENT_OR_OPERATOR,
+  /*    0 */ Token.DIGIT,
+  /*    1 */ Token.DIGIT,
+  /*    2 */ Token.DIGIT,
+  /*    3 */ Token.DIGIT,
+  /*    4 */ Token.DIGIT,
+  /*    5 */ Token.DIGIT,
+  /*    6 */ Token.DIGIT,
+  /*    7 */ Token.DIGIT,
+  /*    8 */ Token.DIGIT,
+  /*    9 */ Token.DIGIT,
+  /*    : */ Token.COLON,
+  /*    ; */ Token.SEMICOLON,
+  /*    < */ Token.OPERATOR,
+  /*    = */ Token.OPERATOR,
+  /*    > */ Token.OPERATOR,
+  /*    ? */ Token.OPERATOR,
+  /*    @ */ Token.AT,
+  /*    A */ Token.IDENTIFIER,
+  /*    B */ Token.IDENTIFIER,
+  /*    C */ Token.IDENTIFIER,
+  /*    D */ Token.IDENTIFIER,
+  /*    E */ Token.IDENTIFIER,
+  /*    F */ Token.IDENTIFIER,
+  /*    G */ Token.IDENTIFIER,
+  /*    H */ Token.IDENTIFIER,
+  /*    I */ Token.IDENTIFIER,
+  /*    J */ Token.IDENTIFIER,
+  /*    K */ Token.IDENTIFIER,
+  /*    L */ Token.IDENTIFIER,
+  /*    M */ Token.IDENTIFIER,
+  /*    N */ Token.IDENTIFIER,
+  /*    O */ Token.IDENTIFIER,
+  /*    P */ Token.IDENTIFIER,
+  /*    Q */ Token.IDENTIFIER,
+  /*    R */ Token.IDENTIFIER,
+  /*    S */ Token.IDENTIFIER,
+  /*    T */ Token.IDENTIFIER,
+  /*    U */ Token.IDENTIFIER,
+  /*    V */ Token.IDENTIFIER,
+  /*    W */ Token.IDENTIFIER,
+  /*    X */ Token.IDENTIFIER,
+  /*    Y */ Token.IDENTIFIER,
+  /*    Z */ Token.IDENTIFIER,
+  /*    [ */ Token.OPENBRACKET,
+  /*    \ */ Token.INVALID,
+  /*    ] */ Token.CLOSEBRACKET,
+  /*    ^ */ Token.OPERATOR,
+  /*    _ */ Token.IDENTIFIER,
+  /*    ` */ Token.TEMPLATELITERAL,
+  /*    a */ Token.MAYBE_KEYWORD,
```

Comment:
I understand the intention here, but still a `MAYBE_KEYWORD` token that is not actually a token seems a little odd. Could this use a different sentinel value?

Reply:
How about `IDENTIFIER_OR_KEYWORD`?